kotlin-eclipse-core/src/org/jetbrains/kotlin/core/preferences/KotlinProperties.kt

```diff
@@ -33,9 +35,37 @@ class KotlinProperties(scope: IScopeContext = InstanceScope.INSTANCE) : Preferen
     }
 
     val compilerPlugins by ChildCollection(::CompilerPlugin)
-    
+
     var compilerFlags by StringPreference()
 
+    val analyzerCompilerFlags: Map<AnalysisFlag<*>, Any?>
+        get() = compilerFlags?.split("\\s+".toRegex())?.mapNotNull { flagString ->
+            flagString.split("=", limit = 2).takeIf { pair ->
+                pair.size == 2
+            }?.let { pair ->
```

Comment:
You can use `(key, value)` or `(flag, state)` instead of `pair` and `[]` operator.